rpc/src/eth.rs

```diff
@@ -241,6 +241,26 @@ impl<B, C, P, CT, BE, H: ExHashT> EthApi<B, C, P, CT, BE, H> where
 		}
 		Ok(None)
 	}
+
+	fn load_transactions(&self, transaction_hash: H256) -> Result<Option<(H256, u32)>> {
+		let mut transactions: Vec<(H256, u32)> = Vec::new();
+		match frontier_consensus::load_transaction_metadata(
+			self.client.as_ref(),
+			transaction_hash,
+		).map_err(|err| internal_err(format!("fetch aux store failed: {:?}", err)))? {
+			Some(metadata) => {
+				for (block_hash, index) in metadata {
+					transactions.push((block_hash, index));
+				}
+			},
+			None => return Ok(None),
+		};
+
+		if transactions.len() == 1 {
```

Comment:
I think we need to remove this check as well?

By the way, there's no need the for loop in L252 any more. The transaction being built and the returned metadata are the same.